WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -881,6 +881,37 @@ extension WordPressAppDelegate {
         UIButton.appearance(whenContainedInInstancesOf: [WPActionBar.self]).tintColor = .primary
         WPActionBar.appearance().barBackgroundColor = .basicBackground
         WPActionBar.appearance().lineColor = .basicBackground
+
+        // Post Settings styles
+
+        /// Post Settings: Main and subscreens
+        UITableView.appearance(whenContainedInInstancesOf: [PostSettingsViewController.self]).tintColor = .editorPrimary
+        UITableView.appearance(whenContainedInInstancesOf: [AztecNavigationController.self]).tintColor = .editorPrimary
+        UISwitch.appearance(whenContainedInInstancesOf: [PostSettingsViewController.self]).onTintColor = .editorPrimary
+
+        /// Post Settings: Date picker
+        if #available(iOS 14.0, *) {
+            let editorSettingsDatePicker = UIView.appearance(whenContainedInInstancesOf: [SchedulingDatePickerViewController.self])
+            editorSettingsDatePicker.tintColor = .editorPrimary
+        }
+
+        /// Post Settings: Tag selection
+        UIView.appearance(whenContainedInInstancesOf: [PostTagPickerViewController.self]).tintColor = .editorPrimary
+
+        /// Post Settings: Categories selection
+        UIView.appearance(whenContainedInInstancesOf: [WPAddPostCategoryViewController.self]).tintColor = .editorPrimary
+        UIView.appearance(whenContainedInInstancesOf: [PostCategoriesViewController.self]).tintColor = .editorPrimary
+
+        /// It's necessary to target PostCategoriesViewController a second time to 'reset'
+        /// the UI element's `tintColor` for use in the app's Site Settings screen.
+        UIView.appearance(whenContainedInInstancesOf: [PostCategoriesViewController.self, WPSplitViewController.self]).tintColor = .primary
```

Comment:
I do not understand what specific issue this addresses or actually address it. Would you please provide steps to replicate the issue and elaborate on how this address it? 

I presume the response to my above question would also apply to the [sibling reset](https://github.com/wordpress-mobile/WordPress-iOS/blob/e034bc6b8833a069f49e09bfd63718aeb3ee6965/WordPress/Classes/System/WordPressAppDelegate.swift#L912-L914) as well, but, if there is nuanced differences, will you please call that out? 

Thanks!

Reply:
@dcalhoun, absolutely, `PostCategoriesViewController` and `WPMediaPickerViewController` are both used in two places in the app.

`PostCategoriesViewController` can be accessed via the _Post Settings_ screen and _Site Settings_ → _Categories_. So, using `(whenContainedInInstancesOf: [PostCategoriesViewController.self])` was used to change the `.tintColor`, it changed it in both screens where the controller is used. 

I couldn't see any other views in the view hierarchy when looking at the categories section of the _Post Settings_ screen. I, therefore, couldn't see a way to _only_ change the `tintColor` for that screen.

I did see some extra views when looking at the hierarchy for the `_Site Settings_ → _Categories_` screen though.  That's what led me to this approach, where I change the color for both screens but then use `(whenContainedInInstancesOf: [PostCategoriesViewController.self, WPSplitViewController.self])` to 'reset' the change to `_Site Settings_ → _Categories_`.

Does that make sense? I agree that this doesn't seem optimal, but think my lack of iOS knowledge is again leading me down hacky paths here.

@twstokes, I wondered if you'd perhaps have ideas for cleaning this up a bit? 🙇‍♀️

Reply:
Aha! Yes, that explanation helps immensely. I now understand the problem and solution. Thank you. 🙇🏻 

Relying upon the view hierarchy appears to be an established pattern for this file. However, the reset additions do feel a bit more brittle, as they appear to rely upon fairly generic hierarchy (e.g. `WPSplitViewController`) rather than more unique hierarchy (e.g. `PostSettingsViewController`) and require a set + reset pattern. 

Unfortunately, I do not have an alternative to propose currently. 🤔 Hopefully others will be able to provide helpful perspective for us.

Reply:
👋 @SiobhyB and @dcalhoun! Thanks for the great discussion and detailed info, they've been really helpful with getting up to speed.

When I inspect the view hierarchy in Xcode with the editor open it seems to me like `AztecNavigationController` is always there. Do you think we could use that? It would have the added benefit of enveloping other views outside of `GutenbergViewController` like `PostSettingsViewController`.

Perhaps the calls could be simplified to:

```
UIView.appearance(whenContainedInInstancesOf: [AztecNavigationController.self]).tintColor = .editorPrimary
UISwitch.appearance(whenContainedInInstancesOf: [AztecNavigationController.self]).onTintColor = .editorPrimary
```

_I only did a quick smoke test so there's a really high chance I left something out or have missed something major._

![AztecNavigationController Hierarchy](https://user-images.githubusercontent.com/2092798/182748284-a5f9d7f8-6bbd-49d1-a718-0bc66e0feaac.jpg)

Reply:
@twstokes, thank you again for helping with this! 🙇‍♀️  Your prompt made me realise there was a redundant call to `UITableView` and I went ahead to refine the `UISwitch` call, as per your suggestion, in https://github.com/wordpress-mobile/WordPress-iOS/pull/19113/commits/88689745d008e54e9a2f00b50bcc318b4156e9d2.

Refactoring the call to `UITableView` to `UIView` led to some unexpected changes in the main editor view. You'll see that the colours in the formatting toolbar, amongst a few other smaller UI elements, were changed when swapping `UITableView` with `UIView`:

| `UITableView.appearance`  | `UIView.appearance` |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/182810302-fbc62297-8d51-4233-98fd-3a3612fd82f4.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/182809729-6edfb852-d030-4154-964e-77043aec4b3e.png" width="100%"> |

In addition, I believe targeting `UIView` instead of `UITableView` here would only prevent the need to target the Post Settings' tag selection screen separately. 

The date picker, featured media selection, and category sections (specifically, the + and parent category subcreens) would still need separate calls. None of these appear to be children of `AztecNavigationController`.

Let me know if there's something I'm missing here.

The more I think about this, the more I wonder whether we're attempting to do something a bit too unusual with the way we're styling different views. I feel pretty iffy about some of the hacks in this PR. If there isn't a cleaner way to approaching some of these challenges, perhaps it'd be preferable to leave the `tintColor` in Post Settings as green and only change the editor's colours? (Though there'd still be some hacks left over to style the editor's colours.)

Looking forward to hearing if you have other ideas for tackling this, too.

Reply:
> You'll see that the colours in the formatting toolbar, amongst a few other smaller UI elements, were changed when swapping UITableView with UIView

Ah, I see! I didn't spot those.

> I believe targeting UIView instead of UITableView here would only prevent the need to target the Post Settings' tag selection screen separately.

That makes sense.

> the more I wonder whether we're attempting to do something a bit too unusual with the way we're styling different views

I think we should be able to avoid the "reset" portions, at least. It's also not super common (in my experience) to set the `tintColor` via the appearance proxy, though there's nothing wrong with that. Typically I'd see it set at the view controller or view level.

I'll look at this some more and report back.

Reply:
@SiobhyB - would this patch be equivalent to the code currently in this section? (i.e. views that are children of `AztecNavigationController`)

**Summary:**
- Removes the resets by increasing specificity
- When there's only one `.tintColor` for an entire `UIView`, it moves the call to the corresponding view controller (it should propagate down to subviews the same way)
    - _Note:_ I didn't spend time finding the optimal spot to set `view.tintColor`, so if you go with this solution feel free to assess that. Typically somewhere in the `viewDidLoad` process works.

```
diff --git a/WordPress/Classes/System/WordPressAppDelegate.swift b/WordPress/Classes/System/WordPressAppDelegate.swift
index 91c7ec65ca..55ac1d4ab2 100644
--- a/WordPress/Classes/System/WordPressAppDelegate.swift
+++ b/WordPress/Classes/System/WordPressAppDelegate.swift
@@ -887,30 +887,8 @@ extension WordPressAppDelegate {
         /// Post Settings: Main and subscreens
         UITableView.appearance(whenContainedInInstancesOf: [AztecNavigationController.self]).tintColor = .editorPrimary
         UISwitch.appearance(whenContainedInInstancesOf: [AztecNavigationController.self]).onTintColor = .editorPrimary
-
-        /// Post Settings: Date picker
-        if #available(iOS 14.0, *) {
-            let editorSettingsDatePicker = UIView.appearance(whenContainedInInstancesOf: [SchedulingDatePickerViewController.self])
-            editorSettingsDatePicker.tintColor = .editorPrimary
-        }
-
-        /// Post Settings: Tag selection
-        UIView.appearance(whenContainedInInstancesOf: [PostTagPickerViewController.self]).tintColor = .editorPrimary
-
-        /// Post Settings: Categories selection
-        UIView.appearance(whenContainedInInstancesOf: [WPAddPostCategoryViewController.self]).tintColor = .editorPrimary
-        UIView.appearance(whenContainedInInstancesOf: [PostCategoriesViewController.self]).tintColor = .editorPrimary
-
-        /// It's necessary to target PostCategoriesViewController a second time to 'reset'
-        /// the UI element's `tintColor` for use in the app's Site Settings screen.
-        UIView.appearance(whenContainedInInstancesOf: [PostCategoriesViewController.self, WPSplitViewController.self]).tintColor = .primary
-
-        /// Post Settings: Featured Image Search
-        UIView.appearance(whenContainedInInstancesOf: [WPMediaPickerViewController.self]).tintColor = .editorPrimary
-
-        /// It's necessary to target WPMediaPickerViewController a second time to 'reset'
-        /// the UI element's `tintColor` for use app's main Media screen.
-        UIView.appearance(whenContainedInInstancesOf: [WPMediaPickerViewController.self, WPSplitViewController.self]).tintColor = .primary
+        UIView.appearance(whenContainedInInstancesOf: [AztecNavigationController.self, PostCategoriesViewController.self]).tintColor = .editorPrimary
+        UIView.appearance(whenContainedInInstancesOf: [AztecNavigationController.self, WPMediaPickerViewController.self]).tintColor = .editorPrimary
     }
 }
 
diff --git a/WordPress/Classes/ViewRelated/Post/Categories/WPAddPostCategoryViewController.m b/WordPress/Classes/ViewRelated/Post/Categories/WPAddPostCategoryViewController.m
index c7a99fb4b8..b8a15c1d1f 100644
--- a/WordPress/Classes/ViewRelated/Post/Categories/WPAddPostCategoryViewController.m
+++ b/WordPress/Classes/ViewRelated/Post/Categories/WPAddPostCategoryViewController.m
@@ -40,6 +40,7 @@ - (void)viewDidLoad
     self.tableView.sectionFooterHeight = 0.0f;
     // Set this to 0 to avoid automattic sizing of cells and getting label cell to short.
     self.tableView.estimatedRowHeight = 0.0f;
+    self.view.tintColor = [UIColor murielEditorPrimary];
 
     self.saveButtonItem = [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"Save", @"Save button label (saving content, ex: Post, Page, Comment, Category).")
                                                            style:[WPStyleGuide barButtonStyleForDone]
diff --git a/WordPress/Classes/ViewRelated/Post/PostTagPickerViewController.swift b/WordPress/Classes/ViewRelated/Post/PostTagPickerViewController.swift
index daecdc2af0..b86bb9f771 100644
--- a/WordPress/Classes/ViewRelated/Post/PostTagPickerViewController.swift
+++ b/WordPress/Classes/ViewRelated/Post/PostTagPickerViewController.swift
@@ -41,6 +41,7 @@ class PostTagPickerViewController: UIViewController {
         WPStyleGuide.configureTableViewColors(tableView: tableView)
 
         view.backgroundColor = .listBackground
+        view.tintColor = .editorPrimary
 
         textView.delegate = self
         // Do any additional setup after loading the view, typically from a nib.
diff --git a/WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingDatePickerViewController.swift b/WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingDatePickerViewController.swift
index 6d75a153ed..ce9a68378a 100644
--- a/WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingDatePickerViewController.swift
+++ b/WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingDatePickerViewController.swift
@@ -73,6 +73,7 @@ class SchedulingDatePickerViewController: UIViewController, DatePickerSheet, Dat
         navigationItem.setRightBarButton(doneButton, animated: false)
 
         setup(topView: chosenValueRow, pickerView: datePickerView)
+        view.tintColor = .editorPrimary
 
         setupForAccessibility()
     }
```

Reply:
@twstokes, unfortunately, I'm seeing the green tint colours in those views when applying the patch:

| Parent Categories  | Media Selection |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/182963243-20564a10-a4d9-4282-a43d-dab19e202167.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/182963162-52dc0cb5-3b91-4b38-b876-07a8c88ade1b.png" width="100%"> |

I don't see `AztecNavigationController.self` in the view hierarchy for either of those views, either. 🤔

Reply:
> I don't see AztecNavigationController.self in the view hierarchy for either of those views, either. 🤔

Good call, I didn't drill down into the Parent Category portion to see that it's not included under  `AztecNavigationController.self`.

I didn't see the same for Media Selection, though. Perhaps we're getting there a different way? I'm adding an image block and selecting "WordPress Media Library". I'm not quite sure how that's working!

<img src="https://user-images.githubusercontent.com/2092798/182967661-606100c7-317d-48b7-9f37-577ac0c032e0.png" width="350px" />

Reply:
> I didn't see the same for Media Selection, though. Perhaps we're getting there a different way? I'm adding an image block and selecting "WordPress Media Library". I'm not quite sure how that's working!

Aha, I see! I can confirm that I'm seeing blue when adding media via the image block, without any extra calls.

The area I'm still seeing the green tint is after navigating to Post Settings, selecting Set Featured Image, and then choosing to select an image from the Media Library. 

I'm wondering if maybe it'd be better to accept the green tint on those two screens (parent category and featured media selection) if there isn't a less hacky alternative.  

(I'll also incorporate your other suggestions for improving the overall code tomorrow, thank you for those!)

Reply:
I've gone ahead to incorporate the suggestions from your patch in https://github.com/wordpress-mobile/WordPress-iOS/pull/19113/commits/56dab966a361844f1e30cdea8ce4d71218d0ae1b. 🙇‍♀️